src/error.rs

```diff
@@ -9,56 +8,48 @@ use std::fmt;
 /// This list is non-exhaustive.
 ///
 /// [`Error`]: std.struct.Error.html
-#[derive(Debug, Fail)]
+#[derive(Debug)]
+#[non_exhaustive]
```

Comment:
Added `non_exhaustive` since we already make a breaking change by removing `failure`